tempodb/blocklist/poller.go

```diff
@@ -456,6 +466,46 @@ func (p *Poller) tenantIndexPollError(idx *backend.TenantIndex, err error) error
 	return nil
 }
 
+// deleteTenant will delete all of a tenant's objects if there is not a tenant index present.
+func (p *Poller) deleteTenant(ctx context.Context, tenantID string) error {
+	level.Info(p.logger).Log("msg", "deleting tenant", "tenant", tenantID)
+
+	if p.cfg.EmptyTenantDeletionAge == 0 {
+		return fmt.Errorf("empty tenant deletion age must be greater than 0")
+	}
+
+	var foundObjects []string
+	err := p.reader.Find(ctx, backend.KeyPath{tenantID}, func(opts backend.FindOpts) {
+		level.Info(p.logger).Log("msg", "checking object for deletion", "object", opts.Key, "modified", opts.Modified)
+
+		if time.Since(opts.Modified) > p.cfg.EmptyTenantDeletionAge {
+			foundObjects = append(foundObjects, opts.Key)
```

Comment:
We could also call delete directly here, but I thought checking once more for the index would be good, which we do below.  Additionally, not having an error return here makes the interface for FindFunc simpler.

Reply:
Agree, separating the search and delete steps seems good, also to avoid any complications from changing the search results while going.  I think it would be good to log what's being deleted.    One more thought:  it looks like are deleting everything older than the cutoff, on each pass. Does it matter if anything newer is found?

Reply:
I'm not sure that it matters if newer data is found, but I think I see what you are getting at.  The reason I think it probably doesn't matter is that the partial block may have been written ages ago.  Would it perhaps be better to only return the values if __all__ data is older than the `p.cfg.EmptyTenantDeletionAge`?  So any more recent data would be a reason to bail on the delete, not just the appearance of the index below.

Reply:
I pushed an additional safety check so that any object with a recent timestamp, we avoid deleting.  This way, all objects are subject to deletion at the same time, only after the most recently modified object crosses the threshold.  Hows that seem to you?

Reply:
Yeah this change seems good.  I was considering that indiscriminately deleting any old files might be dangerous, not under normal circumstances, but situations like a misconfigured or shared bucket, custom workflows, etc.    Not recommending it, but theoretically Tempo could also only delete files it owns like `data.parquet`, `bloom-*`.

Reply:
That's a good thing to keep in mind.  Let's see how it goes without matching the file names.